### PR TITLE
Feat: fetch gateway groups for device info

### PR DESCRIPTION
**Pull request by rkpattnaik780**
### Type of change

- Enhancement

### Description

Display gateway groups and group membership in device info page.

Upgrades to #5120 

### Screenshots

Screenshot from 2020-08-18 16-15-35

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Resolve all warnings
- [ ] Update license
- [ ] Write test cases for null checks in Unit Tests
- [x] Reference relevant issue(s) and close them after merging

### CODE REVIEW

File: console/console-init/ui/src/modules/iot-device-detail/components/GatewayMembership/GatewayMembership.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+              icon={<OutlinedQuestionCircleIcon />}
+            ></Button>
+          </Popover>
+        </Title>
+      </CardTitle>
+
+      <CardBody className={css(styles.flex_item_margin)}>
+        <Flex>
+          <br />
+          {memberOf &&
```
Comment (drcoolsanjeev) [inline console/console-init/ui/src/modules/iot-device-detail/components/GatewayMembership/GatewayMembership.tsx:72]: what happens if memberOf is not present?

Comment (rkpattnaik780) [reply console/console-init/ui/src/modules/iot-device-detail/components/GatewayMembership/GatewayMembership.tsx:72]: Screenshot from 2020-08-18 16-41-51

Comment (drcoolsanjeev) [reply console/console-init/ui/src/modules/iot-device-detail/components/GatewayMembership/GatewayMembership.tsx:72]: Also the color of tooltip icon doesn't seems accurate to me

Comment (suyash-naithani) [reply console/console-init/ui/src/modules/iot-device-detail/components/GatewayMembership/GatewayMembership.tsx:72]: It's the default color for popover provided by patternfly, I would suggest we stick with this color for consistency and discuss it with the UX team

Comment (drcoolsanjeev) [reply console/console-init/ui/src/modules/iot-device-detail/components/GatewayMembership/GatewayMembership.tsx:72]: @rkpattnaik780 This is the design color. I would suggest to add custom styling to change the color of icon 
image

File: console/console-init/ui/src/modules/iot-device-detail/components/GatewayMembership/GatewayMembership.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            memberOf.map((deviceId: string) => {
```
Comment (ajaypratap003) [inline console/console-init/ui/src/modules/iot-device-detail/components/GatewayMembership/GatewayMembership.tsx]: deviceId is not the right name. can we try member or group etc.

File: console/console-init/ui/src/modules/iot-device-detail/components/ConnectionGateway/ConnectionGateway.tsx [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+});
+
+export const ConnectionGateway: React.FC<IConnectionGatewayProps> = ({
+  deviceList,
+  gatewayGroups
+}) => {
+  const { projectname, namespace } = useParams();
+
+  return (
+    <Card className={css(styles.card_body)}>
```
Comment (anukritijha) [inline console/console-init/ui/src/modules/iot-device-detail/components/ConnectionGateway/ConnectionGateway.tsx:41]: Please divide this component in 3 smaller components for Connection-Gateways, Gateway-Devices and Gateway-Groups

File: console/console-init/ui/src/modules/iot-device-detail/components/GatewayMembership/GatewayMembership.tsx (hunk 2 above)
Comment (rkpattnaik780) [reply console/console-init/ui/src/modules/iot-device-detail/components/GatewayMembership/GatewayMembership.tsx]: You have probably commented on a previous commit, I have renamed it to `group`. Will that work?

**Pull request merged**